### PR TITLE
Optimize the check for negative bignums.

### PR DESCRIPTION
**Pull request by LemonBoy**
We save a function call for every check.

### CODE REVIEW

File: src/math/bigintops.c [hunk 1]
```diff
@@ -543,13 +543,13 @@ MVMObject *MVM_bigint_div(MVMThreadContext *tc, MVMObject *result_type, MVMObjec
     bc = get_bigint_body(tc, result);
 
     if (MVM_BIGINT_IS_BIG(ba)) {
-        cmp_a = mp_cmp_d(ba->u.bigint, 0);
+        cmp_a = SIGN(ba->u.bigint) == MP_NEG ? MP_LT : MP_GT;
     } else {
         // we only care about MP_LT or !MP_LT, so we give MP_GT even for 0.
         cmp_a = ba->u.smallint.value < 0 ? MP_LT : MP_GT;
```
Comment (LemonBoy) [inline src/math/bigintops.c]: We could probably make this code use `MP_NEG` and `MP_ZPOS` instead

**Comment by jnthn** [comment]
@LemonBoy any reason to close this besides "too slow to get review"? Finally got around to looking at it now, and it looks fine.

**Comment by LemonBoy** [comment]
None in particular, it got killed in a housekeeping spree :)
You can easily grab the [patch](https://patch-diff.githubusercontent.com/raw/MoarVM/MoarVM/pull/405.patch) and apply it using `git` though, I'd be happy to see this applied.

**Comment by MasterDuke17** [comment]
I seem to recall this was related to https://rt.perl.org/Ticket/Display.html?id=128035, was the patch actually applied?

**Comment by jnthn** [comment]
D'oh, I forgot. It's applied now, though it was a welcome optimization rather than a behavior change, so no impact on the RT in question.

**Pull request closed without merging**